### ISSUE TITLE
not on the classpath of project app, only syntax errors are reported

### ISSUE THREAD

**Issue by tapz** [redhat-developer/vscode-java#1390]
MyCall is not on the classpath of project app, only syntax errors are reported

##### Environment
- Operating System: macOS Catalina 10.15.4 (19E287)
- JDK version: 1.8.0_40
- Visual Studio Code version: 1.44.0
- Java extension version: Java Extension Pack 0.90.0, Language Support for Java 0.59.1

##### Steps To Reproduce
1. Open Java file in vscode

##### Current Result

Error

##### Expected Result

No error

**Comment by fbricon** [comment]
Can you provide more details? like your project setup? maybe share your project?

**Comment by tapz** [comment]
I can't share my project. It's a React Native project. Just a normal Java project, which compiles just fine.

**Comment by fbricon** [comment]
So I'm assuming it's an Android project then? If so, then sorry but it's not supported by [vscode-java ](https://github.com/redhat-developer/vscode-java/issues/10#issuecomment-268834749)

**Comment by tapz** [comment]
Yes. Android. What makes Android so different? It's my Java class after all, not any Android class.

**Comment by fbricon** [comment]
Android uses special Gradle plugins to compute the project classpath, that are not supported in Buildship, the underlying Gradle Integration plugin we delegate Gradle support to.

**Comment by agusfahrurroji** [comment]
@fbricon How do you solve this problem?

**Comment by akaroml** [comment]
> @fbricon How do you solve this problem?

Android development is not supported at the moment. It will require a significant amount of work at Buildship as mentioned by @fbricon 

Currently, there is no ideal solution for that but you can try the Lightweight mode by setting`java.server.launchMode` to `LightWeight`. You will be able to use a subset of features like **Go to Definition** of the types inside your local source and JDK. But there are also limitations, you won't be able to navigate to the types in Android SDK, and you won't be able to build and debug.

**Comment by vikashvk** [comment]
Add working files to the project's source path

**Comment by PPPPzzz** [comment]
I have the same problem when using 'package package_name' , my solution is that clost it and open it again....

**Comment by qqbuby** [comment]
I have the same problem:

1. open a folder with vs code, and create a pom.xml to support multiple modules.
2. create a maven project with Spring Initializer, e.g. demo1
3. now, the new project demo1 will not be recognized for 'xxxx.java is not on the classpath of project xxx, only syntax errors are reported'

- - -

btw, i can clean .classpath file (e.g. `git clean -dfx`) and reopen the folder with vscode to solve it.

**Comment by andremeiras** [comment]
I had the same problem here, but I guess that I managed to solve this, just moving the '.vscode' folder into the project folder. Into that have a JSON file pointing to the project path. When I oppened that I've tried to change the path, but had no success. When I moved the entire folder, and after restart VS Code, the error disapeer. Then, I ran a HelloWorld.java test, and it worked for me.

**Comment by abdeali004** [comment]
In VS Code just right-click on the src folder and select the **" Add folder to Java source path"** option. If it didn't' worked, then try to remove it first by selecting **"remove the folder from Java source path"** and then adding it again.

**Restart** once when done. Hope it works for you.

**Comment by LeoK987** [comment]
> In VS Code just right-click on the _src_ (the folder which contains all java files) folder and select the **" Add folder to Java source path"** option. If it didn't' worked, then try to remove it first by selecting **"remove the folder from Java source path"** and then adding it again.
> 
> **Restart** once when done. Hope it works for you.

Can't add to source path: Unsupported operation. Please use build.gradle file to manage the source directories of gradle project.

**Comment by IAmAnshul07** [comment]
Still it shows 

unsupported operation. please use build.gradle
